src/App.js

```diff
@@ -7,19 +7,47 @@ import { Playlist } from "./pages/Playlist/Playlist";
 import { History } from "./pages/History/History";
 import { LikedVideos } from "./pages/Liked Videos/LikedVideos";
 import { Aside } from "./components/Aside/Aside";
+import { Login } from "./pages/Authentication/Login";
+import { Signup } from "./pages/Authentication/Signup";
+import MockAPI from "./components/Mockman";
+import { Profile } from "./pages/Profile/Profile";
+import { RequiresAuth } from "./components/RequiresAuth";
+import { LandingPage } from "./pages/Landing/LandingPage";
+import { useData } from "./context/data-context";
 
 function App() {
+  const { sideMenu } = useData();
+
   return (
     <div className="App">
       <Navbar />
-      <div className="grid grid-20-80">
-        <Aside />
+      <div className={sideMenu ? "grid grid-20-80" : ""}>
+        {sideMenu && <Aside />}
         <Routes>
-          <Route path="/" element={<Home />} />
-          <Route path="/watchlater" element={<Watchlist />} />
+          <Route path="/" element={<LandingPage />} />
+          <Route path="/home" element={<Home />} />
+          <Route
+            path="/watchlater"
+            element={
+              <RequiresAuth>
+                <Watchlist />
+              </RequiresAuth>
```

Comment:
You can explore having nested private routes, and have a separate handler component to check the route condition